homeassistant/components/onkyo/media_player.py

```diff
@@ -259,8 +284,13 @@ def async_onkyo_connect_callback():
     except Exception:
         raise PlatformNotReady from Exception
 
-    for zone in ["main"] + zones:
-        active_zones.append(OnkyoAVR(avr, name, sources, zone, max_volume))
+    # Discover what zones are available for the avr by querying the power.
+    # If we get a response for the specific zone, it means it is available.
+    for zone in zones:
```

Comment:
Isn't discovery for the case that they're not configured manually?

Reply:
When they are not configured manually, `zones` will default to all possible zones (zone2, zone3, zone4). It will discover if any of those zones are available on the AVR and if so, create an entity for it.
When the zones are configured manually, `zones` will be the list of configured zones. But I think it is still good to check if they are available on the AVR instead of just creating entities without checking if the zones are available.

Reply:
Because we only create the zone entities after we checked that they are available on the AVR, `active_zones` will always only contain the main zone when we call `async_add_entities(active_zones.values())` on line 300. So should I change this to `async_add_entities([main_zone])`?

Reply:
I believe you're referring to my deleted comment, that was a misunderstanding on my part :)

Reply:
Not necessarily, that was about the reconnect callback. But now I mean adding the entities to home assistant.
The main zone is added on line 300: `async_add_entities(active_zones.values())`, and all discovered zones are added on line 256: `async_add_entities([zone_entity])`.
Now I was wondering, since we know that `active_zones.values()` always only contains the main zone at line 300, should I change this to `async_add_entities([main_entity])` or something like that?

Reply:
Oh, yeah, that's a good point just to ensure we're not accidentally adding one of the zones twice.